core/templates/pages/splash-page/splash-page.controller.spec.ts

```diff
@@ -38,8 +40,12 @@ describe('Splash Page', function() {
     $timeout = $injector.get('$timeout');
     $q = $injector.get('$q');
     UserService = $injector.get('UserService');
+    LoaderService = $injector.get('LoaderService');
     SiteAnalyticsService = $injector.get('SiteAnalyticsService');
-
+    LoaderService.getLoadingMessageSubject().subscribe(
```

Comment:
I have a question about using `subscribe` in general. As I remember, we need to unsubscribe it otherwise it will not be removed from memory. Is there any possibility to assign this line to a variable and then unsubscribe it in an `afterEach` block?

I see you're using the `subscribe` in other places than unit file tests. I think they would follow the same logic using the `unsubscribe` method. You should try to fix it in all files in this PR which uses `subscribe`.

Reply:
Thanks for pointing it out. In case of a component, we can just use the $onDestroy method then unsubscribe there. In case of a directive we will have to use `$scope.$on('$destroy', function(){ ...`.

I will make the changes in the spec files. 
I didn't add the unsubscribe lines because it shouldn't be a problem in the first place. Apart from the spec files, I am subscribing to them in the loading-message component and two other services. The services stay in memory as long as the module stays (so, basically till we close the browser tab or navigate to a different module/ chunk). So for the services it is shouldn't be a problem.  As for the loading-message component, it is in the base directive and the base-directive is only created once the MainPage of the module is loaded. So no case of multiple subscriptions here too. If there was a component that was getting destroyed and recreated multiple times in the same module, then using the subscriptions without the unsubscribe would be bad memory management.

Reply:
Thanks for the explanation @srijanreddy98! So adding `unsubscribe` only in the spec files is enough.